### DISCUSSION TITLE
Suggestions and Loading Symbol not loading when created dynamically

### DISCUSSION THREAD

**Discussion by kinjal-thakkar-granicus** [primefaces/community#425] [PrimeNG]
Hi,
While trying to create a component dynamically using createComponent API from @angular/core, the suggestions and loading symbol arent loading.

HTML
<p-autoComplete [style]="{'width':'100%'}" [inputStyle]="{'width':'100%'}" [(ngModel)]="selectedAddress" [suggestions]="cSVSuggestionModelOutput.Attributes" (completeMethod)="csvSuggestionHandler($event)" field="{{infos.searchFieldName}}" [minLength]="3" placeholder="{{infos.placeholder}}" (onSelect)="findAddress()" (onKeyUp)="keyPress($event)">
<ng-template let-address pTemplate="item" #itemRender>

<div [innerHTML]="address[infos.searchFieldName] | highlight: selectedAddress"></div>

</ng-template>
</p-autoComplete>

TS
reloadComponent(defaultValue:any){

const someDOMElement = document.querySelector('.host app-address-lookup') as HTMLElement;

this.componentRef = createComponent(AddresslookupComponent,{environmentInjector:this.appRef.injector,hostElement:someDOMElement});

this.appRef.attachView(this.componentRef.hostView);

this.componentRef.changeDetectorRef.detectChanges();

}

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz example so we can identify the issue clearly?

**Reply by kinjal-thakkar-granicus** [reply]
Hi,

It was tough replicating the issue in stackblitz.

But I have recorded a video for the same.

[https://www.awesomescreenshot.com/video/22660755?key=d9cab0af7fe153b7123a05bc65a658ee](url) 

The suggestions are loaded properly when page is loaded for the first time. But when any setting is changed, the changes are reflected but loading symbol and suggestions do not load even though the API is hit.


**Refresh Component Code**
 
const someDOMElement = document.querySelector(parameters + ' app-address-lookup') as HTMLElement;
      const componentRef = createComponent(AddresslookupComponent,{environmentInjector:appRef.injector,hostElement:someDOMElement});
      appRef.attachView(componentRef.hostView);
      componentRef.changeDetectorRef.detectChanges();

**Reply by cetincakiroglu** [reply]
Provided link is broken, could you please share a stackblitz reproducer or provide something that we can identify the issue. Shared code doesn't help to understand anything of your component. It just indicates that you're trying to create a dynamic component and triggering change detection of it.
